File: code/java/concurrency/01.java
public CompletableFuture&ltSomeOtherResult&gt loadAndCombine(String name1, String name2){
        loadImageAsync(name1)
            .thenAcceptBoth(loadImageAsync(name2)) ((one, two) -> {
                combineImages(deferred1.await(), deferred2.await())
        )}
}

File: code/kotlin/concurrency/01.kt
suspend fun loadAndCombine(name1: String, name2: String): Image =
        coroutineScope {
            val deferred1 = async { loadImage(name1) }
            val deferred2 = async { loadImage(name2) }
            combineImages(deferred1.await(), deferred2.await())
        }